app/views/articles/_image.html.erb

```diff
@@ -1,7 +1,12 @@
-<% if article.image.present? %>
-  <div class="article-image">
-    <%= link_to_if linked,
-                   image_tag(article.image, class: "u-photo", alt: article.image_description),
-                   article.path %>
-  </div>
-<% end %>
+<div class="article-image">
+  <img class="u-photo"
+       alt="<%= article.image_description %>"
+       src="<%= article.article_image.url || article.image %>"
+       <% if article.article_image.url.present? %>
+         sizes="(min-width: 36em) 33.3vw,100vw"
+         srcset="<%= article.article_image.large %>  1024w,
+                 <%= article.article_image.medium %> 640w,
+                 <%= article.article_image.small %>  320w"
```

Comment:
I'll do my best, but responsive images are still pretty new to me.

 <dl>
<dt>src</dt>
<dd>- the fallback if the browser doesn't recognize the <code>srcset</code> attribute</dd>
  <dt>srcset</dt>
  <dd>- takes a comma separated list of the sources we have (in this case, the versions generated by the uploader ruby code). the <code>w</code> is describing the width of the image source, in pixels.</dd>
  <dt>sizes</dt>
  <dd>- tells the browser how many pixels each source <em>needs</em> by describing the final rendered width of the image using CSS length values. Using <code>vw</code> (as opposed to <code>px</code>) allows the image to stretch. 

<br><br>

<b>I actually have a bug here.</b> I should just have `sizes="100vw"` since the header image is always 100%. 

<br>

If the image size changes at different breakpoints in the CSS, you could encode that as <code>sizes="[media query] [length], [media query] [length], etc"</code></dd>
</dl> 

The resource I used to figure this out was this article: [Responsive Images Done Right: A Guide To \<picture\> And srcset][0]

*note: using `<picture>` would be cool for the potential art-direction it enables, but seems like more work and complication to the article workflow than it's worth*

[0]:https://www.smashingmagazine.com/2014/05/responsive-images-done-right-guide-picture-srcset/